src/ViewRendererBundle/Service/ThemeService.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\ViewRendererBundle\Service;
+
+use ChameleonSystem\CoreBundle\Service\CmsConfigDataAccessInterface;
+use ChameleonSystem\CoreBundle\Service\PortalDomainServiceInterface;
+use ChameleonSystem\ViewRenderer\Interfaces\ThemeServiceInterface;
+
+class ThemeService implements ThemeServiceInterface
+{
+    private PortalDomainServiceInterface $portalDomainService;
+    private CmsConfigDataAccessInterface $cmsConfigDataAccess;
+
+    private ?\TdbPkgCmsTheme $themeOverride = null;
+
+    public function __construct(
+        PortalDomainServiceInterface $portalDomainService,
+        CmsConfigDataAccessInterface $cmsConfigDataAccess
+    ) {
+        $this->portalDomainService = $portalDomainService;
+        $this->cmsConfigDataAccess = $cmsConfigDataAccess;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getTheme(?\TdbCmsPortal $portal): ?\TdbPkgCmsTheme
+    {
+        if (null !== $this->themeOverride) {
+            return $this->themeOverride;
+        }
+
+        if (null === $portal) {
```

Comment:
This should match the old logic.

1. if backend mode, return backend theme
2. if portal, return portal theme
3. otherwise, get active portal and return corresponding theme.
4. still nothing, return null

Reply:
(the old logic was: use portal first, if none is present load the active one)

Reply:
it looks unchanged. Please move 
```php
        if (true === $this->requestInfoService->isBackendMode()) {
            return $this->cmsConfigDataAccess->getBackendTheme();
        }
```

to line 44

Reply:
Done